First release (#27)
caravagnalab/nf-evoverse: docs/usage.md

# nf-core/tumourevo: Usage

## :warning: Please read this documentation on the nf-core website: [https://nf-co.re/tumourevo/usage](https://nf-co.re/tumourevo/usage)

> _Documentation of pipeline parameters is generated automatically from the pipeline schema and can no longer be found in markdown files._

## Introduction

**tumourevo** is a workflow to infer a tumour evolution model from whole-genome sequencing (WGS) data.

> Through the analysis of variant and copy-number calls, it reconstructs the evolutionary process leading to the observed tumour genome. Most of the analyses can be done at mutliple levels: single sample, multiple samples from the same patient (multi-region/longitudinal assays), and multiple patients from distinct cohorts.

Reviewer: ```suggestion
Through the analysis of variant and copy-number calls, it reconstructs the evolutionary process leading to the observed tumour genome. Most of the analyses can be done at multiple levels: single sample, multiple samples from the same patient (multi-region/longitudinal assays), and multiple patients from distinct cohorts.
```

Reviewer 2: Typo has been corrected

| `normal_sample` <br /> _Required_          | **Normal sample ID** of each sample. Must match the normal sample ID present in the VCF.                                                                                                                                |
| `vcf` <br /> _Required_                    | Full path to the vcf file.                                                                                                                                                                                              |
| `tbi` <br /> _Required_                    | Full path to the vcf `tabix` index file.                                                                                                                                                                                |
| `cna_caller` <br /> _Required_             | Name of the copy number caller used to generate your data.                                                                                                                                                              |

> | `cna_segments` <br /> _Required_           | Full path to the segmentation files and copy number state from copy-number calling.                                                                                                                                     |

Reviewer: What are the options for this?

Reviewer 2: We have added the options both for cna_caller and vcf_caller, thank you!

## Introduction

**tumourevo** is a workflow to infer a tumour evolution model from whole-genome sequencing (WGS) data.

Through the analysis of variant and copy-number calls, it reconstructs the evolutionary process leading to the observed tumour genome. Most of the analyses can be done at mutliple levels: single sample, multiple samples from the same patient (multi-region/longitudinal assays), and multiple patients from distinct cohorts.

## Samplesheet input

You will need to create a samplesheet with information about the samples you would like to analyse before running the pipeline. Use this parameter to specify its location. It has to be a comma-separated file with 10 columns, and a header row as shown in the examples below.

```bash
--input '[path to samplesheet file]'
```

### Full samplesheet

The samplesheet can have as many columns as you desire, however, there is a strict requirement for the first 10 columns to
match those defined in the table below.

It is recommended to use the absolute path of the files, but a relative path should also work.

> For the joint analysis of multiple samples, a tumour BAM file is required for each sample, such that the number of reads of a private mutation can be retrieved for all the samples thorugh `mpileup`.

Reviewer: Does it take cram files?

Author: It does indeed. We have updated the documentation accordingly.

| `cna_caller`             | Name of the copy number caller used to generate your data. <br /> _Required_                                                                                                                                                              |
| `cna_segments`           | Full path to the segmentation files and copy number state from copy-number calling. <br /> _Required_                                                                                                                                     |
| `cna_extra`              | Full path to files including the ploidy and purity estimate from the copy-number caller. <br /> _Required_                                                                                                                                |
| `cancer_type`            | Tumour type (either `PANCANCER` or one of the tumour type present in the driver table) <br /> _Required_                                                                                                                                  |

> | `tumour_alignment`       | Full path to the tumour bam file. <br /> _Optional_                                                                                                                                                                                       |

Reviewer: Or cram file

Author: Updated.

> | `tumour_alignment_index` | Full path to the tumour bam index file. <br /> _Optional_                                                                                                                                                                                 |

Reviewer: Or crai file

Author: Updated.

An [example samplesheet](../assets/samplesheet.csv) has been provided with the pipeline.

### Pipeline modalities

The tumourevo pipeline supports variant annotation, driver annotation, quality control processes, subclonal deconvolution and signature deconvolution analysis through various tools. It can be used to analyse both single sample experiments and longitudinal/multi-region assays, in which multiple samples of the same patient are avaiable.

> As input, you must provide at least information on the samples, the VCF file from one of the supported callers and the output of one of the supported copy number caller. By default, if multiple samples from the same patient are provided, they will be analysed in a multivariate framework (which affects in particular the subclonal deconvolution deconvolution steps) to retrieve information useful in the reconstruction of the evolutionary process. Depending on the variant calling strategy (single sample or multi sample) and the provided input files, different strategies will be applied.

Reviewer: ```suggestion
As input, you must provide at least information on the samples, the VCF file from one of the supported callers and the output of one of the supported copy number callers. By default, if multiple samples from the same patient are provided, they will be analysed in a multivariate framework (which affects in particular the subclonal deconvolution steps) to retrieve information useful in the reconstruction of the evolutionary process. Depending on the variant calling strategy (single sample or multi sample) and the provided input files, different strategies will be applied.
```

<!-- aggiungi un riassunto di cosa voglia dire single e multi sample (analisi multivariata, soprattutto per subclonal deconv)

> E' possibile usarla sia nel caso di vc multi sample che indipendente -->

Reviewer: Non posso leggerlo

Author: Abbiamo rimosso i commenti in Italiano :)

#### Variant calling

##### 1. Multi-sample variant calling

Modern tools (ie: Platypus and Mutect2) allow to perform variant calling directly in multisample mode. If the VCFs provided as input are already multisample, no additional step is required.

> <!-- If the variant calling had been performed indepentently on each sample from the same patient,

Reviewer: This is commented out?

Author: Removed.

<!-- rephrase better -->
<!-- Si assume che i campioni siano indipendenti e che quindi la subclonal deconv viene svolta cercando popolazioni sottoclonali a lv di singolo campione. Sign deconv si cercano i processi mutagenici comuni in un dataset fatto di elementi indipenti -->

###### Examples

Running the pipeline

```bash
nextflow run nf-core/tumourevo \
 -r <VERSION> \
 -profile <PROFILE> \
 --input <INPUT CSV> \
 --outdir results \
 --tools pyclonevi,mobster,viber,sparsesignature,sigprofiler
```

Minimal input file, two samples from the same patient:

```csv
dataset,patient,tumour_sample,normal_sample,vcf,tbi,cna_segments,cna_extra,cna_caller,cancer_type
dataset1,patient1,sample1,N1,patient1_sample1.vcf.gz,patient1_sample1.vcf.gz.tbi,/CNA/patient1/sample1/segments.txt,CNA/patient1/sample1/purity_ploidy.txt,caller,PANCANCER
dataset1,patient1,sample2,N1,patient1_sample2.vcf.gz,patient1_sample2.vcf.gz.tbi,/CNA/patient1/sample2/segments.txt,CNA/patient1/sample2/purity_ploidy.txt,caller,PANCANCER
```

##### 2. Single sample variant calling

> If the variant calling performed independently on each sample, even if coming from the same patient, you can provide the BAM and BAI files from each tumour sample. In this way, a classical pileup strategy will be used in order to retrieve the depth for all samples of private mutations, in order to correctly perform the subclonal deconvolution analysis.

Reviewer: Think this is what you mean here.
```suggestion
If the variant calling has been performed independently on each sample, even if coming from the same patient, you can provide the BAM and BAI files from each tumour sample. In this way, a classical pileup strategy will be used in order to retrieve the depth for all samples of private mutations, in order to correctly perform the subclonal deconvolution analysis.
```

Author: Updated.

Two of the avaiable tools for subclonal deconvolution, `pyclonevi` and `viber` can by-design be run in multi-sample mode, inferring the subclonal structure of samples. If you add `mobster` to the `--tool` parameter when running the pipeline in this modality, it will be run at first on each individual sample (since the tool does not support at the moment multi-sample analysis) in order to recognize neutral tail mutations and remove them. The mutations data manipulated in this way will then be processed by either `pyclone`, `viber` or both using the multivariate subclonal deconvolution as described before.  -->

#### 3. Filtering data

During the QC step, the pipeline will combine purity, copy number and mutation data to perform quality control on the copy number calls, by applying the [CNAqc algorithm](https://caravagnalab.github.io/CNAqc/). Each segment (for each sample) will be flagged as passing or not the QC, in the given combination of estimated purity and ploidy. According to CNAqc, a badly called segment should be recalled with a different purity estimation, in order to obtain more reliable results.

> After CNAqc quality control, all the segments (coming from samples of the same patient) are used to build a multi-sample CNAqc object, in which a common segmentation is applied. In this way, only those regions that are shared among all samples will be kept in the new object. It is possible to control wheter to include or not in the new segmentation the segments, for each sample, that do not pass the QC test using the `--filter` flag. If it is setted to true, only QC passing segments for each sample will be used to build the mCNAqc object and will then be passed to the subclonal deconvolution steps. This will lead to exclude some regions of the genome and the mutations that sit on it, but should result in more precise analyses. Otherwise, keeping also the segments that do not pass the QC will result in not loosing any mutation but might lead to less precise results in the subclonal deconvolution steps.

Reviewer: ```suggestion
After CNAqc quality control, all the segments (coming from samples of the same patient) are used to build a multi-sample CNAqc object, in which a common segmentation is applied. In this way, only those regions that are shared among all samples will be kept in the new object. It is possible to control whether to include or not in the new segmentation the segments, for each sample, that do not pass the QC test using the `--filter` flag. If this is set to true, only QC passing segments for each sample will be used to build the mCNAqc object and will then be passed to the subclonal deconvolution steps. This will lead to exclude some regions of the genome and the mutations that sit on it, but should result in more precise analyses. Otherwise, keeping the segments that do not pass the QC will result in not losing any mutations, but might lead to less precise results in the subclonal deconvolution steps.
```

Author: Updated.
